### DISCUSSION TITLE
nerdctl container running in nerdctl reports "subnet 10.4.0.0/24 overlaps with another one on this address space"

### DISCUSSION THREAD

**Discussion by ccureau** [containerd/nerdctl#1371] [Q&A]
### Description

Environment: colima running containerd and nerdctl

I have followed the instructions for building and running `nerdctl` in `docker` from the repository README. I do not have `docker` installed locally because I prefer to use `nerdctl` for my container needs. When running `nerdctl` in `nerdctl`, the following error is reported:

`FATA[0000] subnet 10.4.0.0/24 overlaps with other one on this address space`

My goal behind this operation is to use nerdctl in a Kubernetes environment, which I do have working with one caveat - I need to enable legacy iptables to make it work.

### Steps to reproduce the issue

1. `nerdctl build -t nerdctl:latest .`
2. `nerdctl run --rm -it --privileged nerdctl bash`
3. (in nerdctl container) `nerdctl run --rm docker.io/hello-world`

### Describe the results you received and expected

I can't seem to view, add, or change network settings because of the subnet overlap situation.

### What version of nerdctl are you using?

current clone from repository (as of 2022-08-08 08:42 ET)

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

Colima

### Host information

On host:

```
Client:
 Namespace:	default
 Debug Mode:	false

Server:
 Server Version: v1.5.8
 Storage Driver: overlayfs
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Log: json-file
  Storage: native overlayfs
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.109-0-virt
 Operating System: Alpine Linux v3.14
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.845GiB
 Name: colima
 ID: c9e44533-eb54-4cd7-a0c7-6ae7a136ed94
```

From container running on host:

```
Client:
 Namespace:	default
 Debug Mode:	false

Server:
 Server Version: v1.6.8
 Storage Driver: overlayfs
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Log: fluentd journald json-file
  Storage: native overlayfs stargz
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.109-0-virt
 Operating System: Ubuntu 22.04.1 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.845GiB
 Name: 3a5903aac520
 ID: d989f7a6-8d2a-4d1a-8fe8-6212b60ed4e4
```

**Comment by AkihiroSuda** [comment]
A custom CIDR can be specified via `/etc/cni.d/nerdctl-bridge.conflist` (`~/.config/cni/net.d/nerdctl-bridge.conflist` for rootless)

**Comment by astrostl** [comment]
```
user@mac bin % docker run -it --rm ubuntu /bin/bash
FATA[0000] subnet 10.4.0.0/24 overlaps with other one on this address space
```

I found this issue because I receive an identical error message using the latest Rancher Desktop (which bundles `nerdctl`). This used to work well. I have upgraded Rancher Desktop and macOS itself since it last worked. I've already tried a 'factory reset' and uninstall/reinstall of Rancher Desktop. I'm using `containerd` as the runtime.

UPDATE: reverting to Rancher Desktop 1.5.0 from 1.5.1 and performing a second factory reset resolved it for me.
